Server/src/services/authenticationService.ts

```diff
@@ -0,0 +1,142 @@
+     * verified against the existence password hash in the database. Once a match 
+     * is established, the service will request the parameters required to build the
+     * JWT access token. After retrieval, a JWT token is built and sent back to the user
+     * @param userInformation - This params contains only UserName & PassWord
+     */
+    async checkLoginCredentials(userInformation: ILoginInformation): Promise<IResponse> {
+
+        let verifiedEmail: boolean;
+        let savedPasswordHash: string;
+        let accessToken: string;
+
+        try {
+            verifiedEmail = await AuthenticationModel.verifyIfEmailExists(userInformation.email);
+            if (!verifiedEmail)
+                throw new Error();
+        }
+        catch (error) {
+            this.requestResponse.status = "Failure";
+            this.requestResponse.statusCode = 400;
+            this.requestResponse.response = "Email is not in the System or its mispelled. Check Again";
+
+            return this.requestResponse
+        }
+
+        savedPasswordHash = await AuthenticationModel.verifyPassword(userInformation.email);
+
+        if (!(await argon2.verify(savedPasswordHash, userInformation.password))) {
+            this.requestResponse.status = "Failure";
+            this.requestResponse.statusCode = 400;
+            this.requestResponse.response = "Password does not match";
+
+            return this.requestResponse
+        }
+        else {
+            let jwtParams: IJwtParams;
+            try {
+                jwtParams = await AuthenticationModel.obtainJWTParams(userInformation.email);
+                accessToken = await this.generateJwtToken(jwtParams);
+            } catch (error) {
+                this.requestResponse.status = "Failure";
+                this.requestResponse.statusCode = 500;
+                this.requestResponse.response = "Internal Server Issue. Please try again later";
+                return this.requestResponse
+            }
+
+            this.requestResponse.status = "Success";
+            this.requestResponse.statusCode = 200;
+            this.requestResponse.response = accessToken;
+        }
+
+        return this.requestResponse;
+    }
+    /**
+     * Resposible for hashing the user input password
+     * @param password - user password
+     */
+    private async hashPassword(userPassword: string): Promise<string> {
+
+        let hashedPassword: string;
+        hashedPassword = await argon2.hash(userPassword);
+        return hashedPassword;
+    }
+
+    /**
+     * Resposible for building JWT token. This token contains a unique signature
+     * that is only known the backend server. When requests come into the application
+     * each request will contain an access token and its signature will be verified
+     * @param jwtParams Parameters used to build JWT Token
+     */
+    private async generateJwtToken(jwtParams: IJwtParams): Promise<string> {
```

Comment:
nice, we might wanna also add a renew token method

Reply:
I looked into this. And there were a handful of reasons not to use refresh tokens on a web application without an authentication server. Instead I proposed using access tokens with a 3-6 h time-limit. After which, the user would need to login again.

Reply:
ok good, could you create a confluence page linked to your issue about this?